Automattic/newspack-listings
# feat: child and related listings UI

### All Submissions:






### Changes proposed in this Pull Request:



* Removes the placeholder front-end display of parent listings (to be designed and handled in a separate PR, perhaps using a block-based UI)
* Replaces the sidebar UI for managing parent and child listings with a modal UI using the new `AutocompleteWithSuggestions` component from `newspack-components` ~(note: requires https://github.com/Automattic/newspack-plugin/pull/975 to be published to NPM)~
* As discussed with @thomasguillot, we no longer show parent listings UI in listings (to avoid showing both parent and child UIs in the same editor instance). Parent/child relationships for listings are now solely managed as child listings from the parent's editor instance.

### How to test the changes in this Pull Request:

1. ~Until https://github.com/Automattic/newspack-plugin/pull/975 is published to NPM, spoof the change by checking out that branch in `newspack-plugin`, going to `assets/components`, and running `npm run prepublishOnly`, then copying the `components` folder to `newspack-listings/node_modules/newspack-components`~ This is no longer required as the updated component has been published to NPM. You should be able to run `npm ci` from this branch to get it, now.
2. Edit a post or page. Confirm the new sidebar panel:

<img width="186" alt="Screen Shot 2021-05-25 at 5 11 21 PM" src="https://user-images.githubusercontent.com/2230142/119579707-3ce27300-bd7c-11eb-9862-986591c903d8.png">

3. Click "Manage Related Listings" and confirm the new modal UI:

<img width="566" alt="Screen Shot 2021-05-25 at 5 12 18 PM" src="https://user-images.githubusercontent.com/2230142/119579772-60a5b900-bd7c-11eb-9fdc-d7397f84acb1.png">

4. Select different post types in the dropdown, add some listings, remove others, and save. Confirm that all functionality works as expected.
5. After saving, edit one of the listings you added in WP admin.
6. In the editor page for the listing, confirm the new sidebar panel:

<img width="181" alt="Screen Shot 2021-05-25 at 5 13 51 PM" src="https://user-images.githubusercontent.com/2230142/119579877-96e33880-bd7c-11eb-8fce-faee0462a62e.png">

7. Click "Manage Child Listings" and confirm the new modal UI. Confirm that the post where you selected this listing as a related listing is shown as selected here:

<img width="566" alt="Screen Shot 2021-05-25 at 5 14 14 PM" src="https://user-images.githubusercontent.com/2230142/119579898-a5315480-bd7c-11eb-972f-f53bed6a4452.png">

8. Select different post types in the dropdown, add some listings, remove others, and save. Confirm that all functionality works as expected.
9. Remove the post or page where you added the listing as a related listing in step 4. Save.
10. Go back to that post in the editor, open the "Manage Related Listings" modal and confirm that the listing is no longer shown as selected.
11. Test with different combinations of listings owning other listings, pages, and posts. Confirm that for all listing types, listings can no longer be assigned as a parent to other listings of the same type (they can on the back-end, but only so we can programmatically manage orphaned or missing shadow terms—e.g. every Place listing is assigned its own shadow term, so we can easily query them)

### Other information:
